src/openforms/js/components/admin/form_design/form-creation-form.js

```diff
@@ -524,13 +577,21 @@ function reducer(draft, action) {
     }
     case 'DELETED_RULE': {
       const {index} = action.payload;
-      const ruleUuid = draft.logicRules[index].uuid;
+      const {uuid: ruleUuid, order: ruleOrder} = draft.logicRules[index];
       draft.logicRulesToDelete.push(ruleUuid);
 
       // delete object from state
       const updatedRules = [...draft.logicRules];
       updatedRules.splice(index, 1);
       draft.logicRules = updatedRules;
+
+      // update the rules above this rule
```

Comment:
Maybe a bit clearer?
```suggestion
      // update the rules after this rule
```

Reply:
I picked the same terminology as used in django-ordered-model for consistency - after/before depends a bit if you order ASC or DESC. This makes more sense though, so I'll change it.